discovery_engine_core/core/src/stack.rs

```diff
@@ -162,6 +162,10 @@ impl Bucket<Document> for Stack {
     fn pop(&mut self) -> Option<Document> {
         self.data.documents.pop()
     }
+
+    fn retain_top(&mut self, keep: usize) {
```

Comment:
We use directly the implementation from `self.data` so we don't need to expose this.